app/js/fieldhistorycount.js

```diff
@@ -0,0 +1,22 @@
+var BfsFieldHistoryCount = BfsFieldHistoryCount || {};
+
+BfsFieldHistoryCount.init = function() {
+  $('#ep td[id$="ButtonRow"]').each(function(i, el) {
+    $(el).append('<span class="bfs-history-count">' + BfsFieldHistoryCount.getMessage() + '</span>');
+  });
+  $('#ep table.detailList td.dataCol [type="checkbox"]').change(this.updateCount);
+  $('#ep table.detailList tr.last.detailRow a').click(function() {setTimeout(function(){BfsFieldHistoryCount.updateCount();}, 100);});
```

Comment:
@mattsimonis  This was because the Deselect All Fields link executes javascript which sets the checked property to false. The change event doesn't fire for properties so I went the route of using a timeout.